src/Microsoft.DotNet.Wpf/src/PresentationCore/System/Windows/Media/MediaPlayerState.cs

```diff
@@ -877,7 +877,7 @@ private void OpenMedia(Uri source)
                 {
                     // target is HTTPS. Then, elevate ONLY if we are NOT coming from HTTPS (=XDomain HTTPS app to HTTPS media disallowed)
                     Uri appDeploymentUri = SecurityHelper.ExtractUriForClickOnceDeployedApp();
-                    if (!SecurityHelper.AreStringTypesEqual(appDeploymentUri.Scheme, Uri.UriSchemeHttps))
+                    if (appDeploymentUri != null && !SecurityHelper.AreStringTypesEqual(appDeploymentUri.Scheme, Uri.UriSchemeHttps))
```

Comment:
In .NET Core, it seems to me that the method can probably boil down to something like shown below, and `DemandPermissions(Uri)` just gets deleted. /cc @ojhad , @rladuca thoughts? 

I didn't walk through `MILMedia.Open` carefully to see whether `null` would be always ok, but I'd imagine that would be the case... 

```C#
        private void OpenMedia(Uri source)
        {
            if (source != null && source.IsAbsoluteUri && source.Scheme == PackUriHelper.UriSchemePack)
            {
                try
                {
                    source = BaseUriHelper.ConvertPackUriToAbsoluteExternallyVisibleUri(source);
                }
                catch (InvalidOperationException)
                {
                    source = null;
                    _mediaEventsHelper.RaiseMediaFailed(new System.NotSupportedException(SR.Get(SRID.Media_PackURIsAreNotSupported, null)));
                }
            }

            // Setting a null source effectively disconects the MediaElement.
            if (source != null)
            {

                // get the base directory of the application
                Uri appBase = new Uri(AppDomain.CurrentDomain.BaseDirectory);
                // this extracts the URI to open
                Uri uriToOpen = ResolveUri(source, appBase);				
            }

            HRESULT.Check(MILMedia.Open(_nativeMedia, null));
        }
```